scripts/create_deprecated_modules.lean

```diff
@@ -0,0 +1,378 @@
+-/
+def processPrettyOneLine (log msg fname : String) : IO (String × MessageData) := do
+  let hash := log.takeWhile (!·.isWhitespace)
+  let PRdescr := (log.drop hash.length).trim
+  let gitDiffCLI := s!"git diff {hash}^...{hash} -- {fname}"
+  let diff ← runCmd gitDiffCLI <|> pure s!"{hash}: Error in computing '{gitDiffCLI}'"
+  let diffCollapsed := .trace {cls := .str .anonymous s!"{hash}"} m!"{gitDiffCLI}" #[m!"{diff}"]
+  return (hash, m!"{msg} in " ++
+    .trace {cls := .str .anonymous ("_" ++ hash.take 7)} m!"{PRdescr}" #[diffCollapsed])
+
+/--
+`mkRenamesDict pct` takes as optional input a natural number.
+
+It computes the `git` status of the files at the current `HEAD` commit,
+comparing them with `master`.
+
+It returns a `HashMap` with keys the old names and values the new names of all the files that
+git considers renames with likelihood at least the input `pct`.
+
+If no input is provided, the default percentage is `100`.
+-/
+def mkRenamesDict (percent : Nat := 100) : IO (Std.HashMap String String) := do
+  let mut dict := ∅
+  let gitDiff ← runCmd s!"git diff --name-status origin/master...HEAD"
+  let lines := gitDiff.trim.splitOn "\n"
+  for git in lines do
+    -- If `git` corresponds to a rename, it contains `3` segments, separated by a
+    -- tab character (`\t`): `R%%`, `oldName`, `newName`.
+    let [pct, oldName, newName] := git.split (· == '\t') | continue
+    if pct.take 1 != "R" then
+      IO.println
+        s!"mkRenamesDict: '{pct}' should have been of the form Rxxx, denoting a `R`ename \
+          and a similarity percentage.\nFull git line: '{git}'"
+      continue
+    let pctNat := (pct.drop 1).toNat?.getD 0
+    -- This looks like a rename with a similarity index at least as big as our threshold:
+    -- we add the rename to our dictionary.
+    if percent ≤ pctNat then
+      dict := dict.insert oldName newName
+    -- This looks like a rename, but the similarity index is smaller than our threshold:
+    -- we report a message and do not add the rename to our dictionary.
+    else
+      IO.println
+        s!"'{oldName}' was renamed to '{newName}' ({pct}), but the similarity {pctNat}% \
+          is less than the expected threshold of {percent}%.\n\n
+          We treat this file as a removal."
+  return dict
+
+/--
+`mkModName fname` takes as input a file path and returns the guessed module name:
+the dir-separators get converted to `.` and a trailing `.lean` gets removed, if it exists.
+
+*Note*. The input path is relative to the root of the project.  E.g., within `mathlib`, every path
+effectively starts as `Mathlib/...`.
+-/
+def mkModName (fname : System.FilePath) : String :=
+  let cpts := fname.components
+  let cpts :=
+    match cpts.getLast? with
+    | none => cpts
+    | some last =>
+      cpts.dropLast ++ [if last.endsWith ".lean" then last.dropRight ".lean".length else last]
+  ".".intercalate cpts
+
+#guard mkModName ("Mathlib" / "Data" / "Nat" / "Basic.lean") == "Mathlib.Data.Nat.Basic"
+
```

Comment:
Can you add a unit test for an empty path, just to please the paranoid reviewers? (The current code should be fine.)